CloudWatch: Make it possible to specify custom api endpoint (#31402)
grafana/grafana: docs/sources/datasources/cloudwatch.md

The `Assume Role ARN` field allows you to specify which IAM role to assume, if any. When left blank, the provided credentials are used directly and the associated role or user should have the required permissions. If this field is non-blank, on the other hand, the provided credentials are used to perform an [sts:AssumeRole](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html) call.

### Endpoint

> The `Endpoint` field allows you to specify a custom endpoint URL that overrides the default generated endpoint for the CloudWatch API. Leave this field blank to use the default generated endpoint. Read about why and how to use Service endpoints in the [official AWS documentation](https://docs.aws.amazon.com/general/latest/gr/rande.html).

Reviewer: ```suggestion
The `Endpoint` field allows you to specify a custom endpoint URL that overrides the default generated endpoint for the CloudWatch API. Leave this field blank if you want to use the default generated endpoint. For more information on why and how to use Service endpoints in the [official AWS documentation](https://docs.aws.amazon.com/general/latest/gr/rande.html).
```

> The `Endpoint` field allows you to specify a custom endpoint URL that overrides the default generated endpoint for the CloudWatch API. Leave this field blank if you want to use the default generated endpoint. For more information on why and how to use Service endpoints in the [official AWS documentation](https://docs.aws.amazon.com/general/latest/gr/rande.html).

Author: Oh yes, good catch.

> The `Endpoint` field allows you to specify a custom endpoint URL that overrides the default generated endpoint for the CloudWatch API. Leave this field blank if you want to use the default generated endpoint. For more information on why and how to use Service endpoints, see the [official AWS documentation](https://docs.aws.amazon.com/general/latest/gr/rande.html).

Reviewer 2: ```suggestion
The `Endpoint` field allows you to specify a custom endpoint URL that overrides the default generated endpoint for the CloudWatch API. Leave this field blank if you want to use the default generated endpoint. For more information on why and how to use Service endpoints, refer to the [AWS service endpoints](https://docs.aws.amazon.com/general/latest/gr/rande.html).
```
Conform to style guide, which I am about to update to include this advice.

Reviewer 3: Shouldn't it instead be something like "the AWS service endpoints documentation"? "refer to the AWS service endpoints." doesn't make sense to me language wise.

Author: I agree. Text updated

Reviewer 2: Typically, we default to WYSIWYG when referring out to documentation. Try to reword such that you can use the verbatim title of the referred to document.

Reviewer 3: It reads fine as it is though.
